src/xattrs.c

```diff
@@ -37,6 +37,14 @@ enum xattrs_action_type_t_ {
 
 typedef enum xattrs_action_type_t_ xattrs_action_type_t;
 
+/* If SWUPD_WITH_XATTRS is not defined, create functions to override
+ * glibc's interface to the system calls
+ */
+#ifndef SWUPD_WITH_XATTRS
+#define lgetxattr(p,n,b,l) (-1)
+#define llistxattr(p,b,l) (0)
```

Comment:
The problem is that extended attributes are essentially a key/value store, with the value being a binary blob. Therefore you can have a zero length value,  (This is a space efficient way of storing binary data, just see if the key is defined, but don't store any data).

The lgetxattr call takes a filename and an attribute name, fills in a user provided buffer and returns the length of the attribute value.

Having said that, currently the only way that lgetxattr is called are the two calls inside xattr_get_value (which incidentally has a race condition if the attributes are being changed at the same time as this thing tries to read them). xattr_get_value is only called from xattrs_do_action, and only then if llistxattr gave it some names of attributes to retrieve. Since we say there are never any names, we never will call lgetxattr. The (-1) is just in there for defensive programming, and because I didn't want to start adding in more headers (errno.h).